hbook/h1d.go

```diff
@@ -186,6 +186,36 @@ func (h *H1D) Scale(factor float64) {
 	h.Binning.scaleW(factor)
 }
 
+// AddH1D returns the bin-by-bin summed histogram of h1 and h2 
+// assuming their statistical uncertainies are uncorrelated.
+func AddH1D(h1, h2 *H1D) *H1D {
+	
+	if h1.Len() != h2.Len() {
+		panic("hbook: h1 and h2 have different number of bins")
+	}
+
+	if h1.XMin() != h2.XMin() {
+		panic("hbook: h1 and h2 have different Xmin")
+	}
+
+	if h1.XMax() != h2.XMax() {
+		panic("hbook: h1 and h2 have different Xmax")
+	}
+
+	hsum := NewH1D(h1.Len(), h1.XMin(), h1.XMax())
+	for i := 0; i < hsum.Len(); i++ {
+		x := hsum.Binning.Bins[i].Range.Min
+		y := h1.Value(i) + h2.Value(i)
+		y1err := h1.Binning.Bins[i].SumW2()
+		y2err := h2.Binning.Bins[i].SumW2()
+		yerr := math.Sqrt( y1err*y1err + y2err*y2err) 
+		hsum.Binning.fill(x, y)
+		_ = yerr // Needs something like bin.SetSumW2(x, yerr)
```

Comment:
as this file is part of the `hbook` package, you can reach into all types that are defined inside that package.
so you can actually access all the `sumw2` fields.
_ie:_ access the `Dist` field of the `Binning1D` type, the `Dist Dist1D` field of `Bin1D` (and the various fields there).

Reply:
ok, I gave a shot and it seems I managed to find the proper "field chain", but:
```go
hsum.Binning.Bins[i].Dist.SumW = y
hsum.Binning.Bins[i].Dist.SumW2 = yerr 
```
leads to:
```
# go-hep.org/x/hep/hbook
hbook/h1d.go:211:34: cannot assign to hsum.Binning.Bins[i].Dist.SumW
hbook/h1d.go:212:35: cannot assign to hsum.Binning.Bins[i].Dist.SumW2
```

Reply:
OK found it (it turned out I *didn't* find the proper field chain ...)